presto-native-execution/presto_cpp/main/http/tests/HttpJwtBenchmarkTest.cpp

```diff
@@ -0,0 +1,129 @@
+
+    // Don't use https
+    auto server = getServer(false);
+    server->registerGet("/ping", ping);
+    auto wrapper = std::make_unique<HttpServerWrapper>(std::move(server));
+    auto serverAddress = wrapper->start().get();
+
+    HttpClientFactory clientFactory;
+    auto client = clientFactory.newClient(
+        serverAddress, std::chrono::milliseconds(1'000), false, memoryPool);
+    suspender.dismiss();
+
+    auto response = sendGet(client.get(), "/ping").get();
+
+    suspender.rehire();
+    wrapper->stop();
+    suspender.dismiss();
+  }
+
+  void jwtRun() {
+    folly::BenchmarkSuspender suspender;
+    auto memoryPool = defaultMemoryManager().addLeafPool("basicJwtTest");
+
+    const std::string kSecret{"mysecret"};
+    const std::string kNodeId{"testnode"};
+
+    std::unordered_map<std::string, std::string> systemValues{
+        {std::string(SystemConfig::kMutableConfig), std::string("true")},
+        {std::string(SystemConfig::kInternalCommunicationJwtEnabled),
+         std::string("true")},
+        {std::string(SystemConfig::kInternalCommunicationSharedSecret),
+         kSecret}};
+    std::unique_ptr<Config> rawSystemConfig =
+        std::make_unique<core::MemConfigMutable>(systemValues);
+
+    auto systemConfig = SystemConfig::instance();
+    systemConfig->initialize(std::move(rawSystemConfig));
+
+    std::unordered_map<std::string, std::string> nodeValues{
+        {std::string(NodeConfig::kMutableConfig), std::string("true")},
+        {std::string(NodeConfig::kNodeId), std::string(kNodeId)}};
+    std::unique_ptr<Config> rawNodeConfig =
+        std::make_unique<core::MemConfigMutable>(nodeValues);
+    auto nodeConfig = NodeConfig::instance();
+    nodeConfig->initialize(std::move(rawNodeConfig));
+
+    // Don't use https
+    auto server = getServer(false);
+
+    server->registerGet("/ping", ping);
+
+#ifdef PRESTO_ENABLE_JWT_TOKEN
+    std::vector<std::unique_ptr<proxygen::RequestHandlerFactory>> filters;
+    filters.push_back(std::make_unique<http::filters::JWTTokenFilterFactory>());
+#endif
+    auto wrapper = std::make_unique<HttpServerWrapper>(std::move(server));
+#ifdef PRESTO_ENABLE_JWT_TOKEN
+    wrapper->setFilters(filters);
+#endif
+    auto serverAddress = wrapper->start().get();
+
+    HttpClientFactory clientFactory;
+    auto client = clientFactory.newClient(
+        serverAddress, std::chrono::milliseconds(1'000), false, memoryPool);
+    suspender.dismiss();
+
+    auto response = sendGet(client.get(), "/ping").get();
+
+    suspender.rehire();
+    wrapper->stop();
+    suspender.dismiss();
+  }
+};
+
+std::unique_ptr<HttpJwtBenchmark> benchmark;
+
+template <typename Func>
+void run(Func&& func, size_t iterations = 10000) {
+  for (auto i = 0; i < iterations; i++) {
+    func();
+  }
+}
+
+BENCHMARK(nojwt) {
+  run([&] { benchmark->nojwtRun(); });
+}
+
+BENCHMARK_RELATIVE(jwt) {
+  run([&] { benchmark->jwtRun(); });
+}
+} // namespace
+
+int main(int argc, char* argv[]) {
```

Comment:
Benchmarks should be in a benchmark folder.
We need to add a new option to build benchmarks similar to Velox.